app/services/integration/kubernetes_service.rb

```diff
@@ -6,10 +6,14 @@ class Integration::KubernetesService < Integration::ServiceBase
   class_attribute :maintain_tls_spec,
                   default: ActiveModel::Type::Boolean.new.cast(ENV['KUBERNETES_ROUTE_TLS'])
 
+  attr_reader :use_openshift_route
+
   def initialize(integration, namespace: self.class.namespace)
     super(integration)
     @namespace = namespace
     @client = K8s::Client.autoconfig(namespace: namespace).extend(MergePatch)
+
+    @use_openshift_route = client.api_groups.include?('route.openshift.io/v1')
```

Comment:
> Shouldn't we better perform this call on demand instead of during initialization?
> 
> ```ruby
> def use_openshift_route?
>   @use_openshift_route ||= client.api_groups.include?('route.openshift.io/v1')
> end
> ```

@akostadinov well I had assumed that the environment wouldn't change. I can't imagine a scenario where zync would be running in kubernetes and then the openshift API `route.openshift.io/v1` would be added after. So I figured once at init time would be sufficient. Or am I misunderstanding what you mean?

Reply:
The point is not to change value later. The ``||=` operator will prevent that in either case. The point is not to do operations at object initialization, rather at the time that operation is needed.

At present it may not make much difference. But as a matter of good design. Also one can mock a method much more easily in testing.

Reply:
I updated this, but I'm still not sure I understand the correct way to do this in Ruby classes. I kept having issues using `@use_openshift_route` so I just changed it to return the value. I also added an env variable to force native ingresses because it was simple enough to add it right there.